pyomo/common/numeric_types.py

```diff
@@ -194,6 +203,48 @@ def RegisterLogicalType(new_type: type):
     nonpyomo_leaf_types.add(new_type)
 
 
+def check_if_native_type(obj):
+    if isinstance(obj, (str, bytes)):
+        native_types.add(obj.__class__)
+        return True
+    if check_if_logical_type(obj):
+        return True
+    if check_if_numeric_type(obj):
+        return True
+    return False
+
+
+def check_if_logical_type(obj):
+    """Test if the argument behaves like a logical type.
+
+    We check for "numeric types" by checking if we can add zero to it
+    without changing the object's type, and that the object compares to
+    0 in a meaningful way.  If that works, then we register the type in
+    :py:attr:`native_numeric_types`.
```

Comment:
This docstring seems like it was copied from `check_if_numeric_type` and never updated.